### PR TITLE
Basic specification parser

### PR DESCRIPTION
**Pull request by yourheropaul**

### CODE REVIEW

File: specification/reader.go [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package specification
+
+// A Reader represents the input for a specification. The read method
+// returns a Specification, zero or more warnings, and a fatal error.
+type Reader interface {
+	Read() (*Specification, []error, error)
```
Comment (adrianduke) [inline specification/reader.go]: We've got a `[]error` type already `errors.Errors`, probably worth using here. In fact how about:

```go
type Warning interface {
    error
}
```

the `errors.Errors` satisfies the `error` interface, so could become a `Warning`. Nice and explicit even without comments.

File: specification/reader_filesystem.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+}
+
+// Read reads a specification from disk, returning the spec, any warnings, and
+// possibly a fatal error.
+func (f *Filesystem) Read() (*Specification, []error, error) {
+	spec := NewSpecification()
+	spec.Source = f.Path
+	warnings := []error{}
+
+	err := afero.Walk(f.Fs, f.Path, func(path string, info os.FileInfo, err error) error {
```
Comment (adrianduke) [inline specification/reader_filesystem.go]: Maybe extract out the anonymous func? Should reduce the method size then.

File: specification/specification.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+		lookup[f.trimSource(source)] = source
+	}
+
+	finalSources := []string{}
+	for fs := range lookup {
+		finalSources = append(finalSources, fs)
+	}
+
+	sort.Strings(finalSources)
+	cm := closestmatch.New(finalSources, []int{2, 3})
```
Comment (adrianduke) [inline specification/specification.go]: I don't know what `[]int{2,3}` is, maybe make a named pkg variable?

File: specification/reader.go (hunk 1 above)
Comment (yourheropaul) [reply specification/reader.go]: oh yeah, good idea.

File: specification/reader_filesystem.go [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+	spec.StorySources[path] = story
+
+	return nil
+}
+
+func (f *Filesystem) parseFeatureFile(fs afero.Fs, path string) (*Story, error) {
+	content, err := afero.ReadFile(fs, path)
+
+	if err != nil {
+		// FIXME! Custom error for warnings?
```
Comment (adrianduke) [inline specification/reader_filesystem.go]: We now have a custom warning type, update?

Comment (yourheropaul) [reply specification/reader_filesystem.go]: Already done!

**Pull request merged**